examples/BCDR/BackupFromStorageQueue.cs

```diff
@@ -0,0 +1,172 @@
+    9. Delete all messages that we received in this batch from the storage queue.
+    10. Repeat from step 4 if there are any messages remaining in storage queue.
+ *
+ *
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Extensions.Logging;
+using Azure.Storage.Queues;
+using Azure;
+using Azure.Data.AppConfiguration;
+using Azure.Identity;
+using System.Linq;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json;
+using Azure.Storage.Queues.Models;
+
+namespace BackupFromStorage
+{
+    public static class BackupFromStorageQueue
+    {
+        private static HashSet<KeyLabel> ExtractKeyLabelsFromEvents(QueueMessage[] messages, ILogger log)
+        {
+            HashSet<KeyLabel> updatedKeyLabels = new HashSet<KeyLabel>(new KeyLabelComparer());
+            foreach (QueueMessage message in messages)
+            {
+                try
+                {
+                    // Event grid will encode events in Base64 format before publishing to storage queue
+                    // Decode the message before parsing it
+                    string decodedMessage = Encoding.UTF8.GetString(Convert.FromBase64String(message.MessageText));
+                    try
+                    {
+                        JObject eventObject = JObject.Parse(decodedMessage);
+                        string topic = eventObject.Property("topic")?.Value.ToString();
+                        if (topic != null && topic == Environment.GetEnvironmentVariable("PrimaryStoreResourceId"))
+                        {
+                            JObject dataObject = JObject.Parse(eventObject.Property("data")?.Value.ToString());
+                            string key = dataObject["key"].ToString();
+                            string label = dataObject["label"]?.ToString();
+                            updatedKeyLabels.Add(new KeyLabel(key, label));
+                        }
+                    }
+                    catch (JsonReaderException)
+                    {
+                        // If its not a valid JSON, ignore the message 
+                        log.LogInformation($"Queue message in invalid JSON format will be ignored.\nMessage: {decodedMessage}");
+                    }
+                }
+                catch (FormatException)
+                {
+                    // If its not in valid base64 format, ignore the message 
+                    log.LogInformation($"Queue message in invalid Base64 format will be ignored.\nMessage: {message.MessageText}");
+                }
+            }
+            return updatedKeyLabels;
+        }
+
+        private static void BackupKeyValues(HashSet<KeyLabel> updatedKeyLabels, ILogger log)
+        {
+            // Connect to Azure App Configuration stores
+            ConfigurationClient primaryAppConfigClient = new ConfigurationClient(new Uri(Environment.GetEnvironmentVariable("PrimaryStoreEndpoint")), new ManagedIdentityCredential());
```

Comment:
To make it clearer and avoid reading environment variables again and again, I would define `PrimaryStoreEndpoint` (and others) all as const at the beginning of the class.

Reply:
Done. Also made both clients static, and null-checking them when function is triggered.